src/plugins/csp_configuration_provider/server/csp_handlers.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { CoreSetup, OnPreResponseHandler, OpenSearchClient } from '../../../core/server';
+import { CspClient } from './types';
+
+const OPENSEARCH_DASHBOARDS_CONFIG_DOCUMENT_NAME = 'csp.rules';
+
+export function createCspRulesPreResponseHandler(
+  core: CoreSetup,
+  dynamicConfigIndex: string,
+  getCspClient: (inputOpenSearchClient: OpenSearchClient) => CspClient
+): OnPreResponseHandler {
+  return async (request, response, toolkit) => {
+    const shouldCheckDest = ['document', 'frame', 'iframe', 'embed', 'object'];
+
+    const currentDest = request.headers['sec-fetch-dest'];
+
+    if (!shouldCheckDest.includes(currentDest)) {
+      return toolkit.next({});
+    }
+
+    const [coreStart] = await core.getStartServices();
+
+    const myClient = getCspClient(coreStart.opensearch.client.asInternalUser);
+
+    const existsValue = await myClient.exists(dynamicConfigIndex);
+
+    if (!existsValue) {
+      return toolkit.next({});
+    }
+
+    const cspRules = await myClient.get(
+      dynamicConfigIndex,
+      OPENSEARCH_DASHBOARDS_CONFIG_DOCUMENT_NAME
+    );
```

Comment:
In case of non-opensearch client (e.g. dynamodb, postgres etc), `dynamicConfigIndex` would be table name and `OPENSEARCH_DASHBOARDS_CONFIG_DOCUMENT_NAME` would be key or column, right? If my understanding is correct, we are kinda forcing implementation to follow same naming. Can we make more flexible to check whether dynamic configuration is available or not or get csp rule irrespective of table name or key name? Let plugin decide how they want to store dynamic configuration. If needed we can do validation in core OSD to check CSP rule is passing validation (maybe using regex).

Reply:
Let me make the interface generic and irrespective of table name or key name